### PR TITLE
HOSTEDCP-1933: Expose AWS Tenancy through NodePool API

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
This allows launching AWS EC2 instances into dedicated instances or dedicated hosts.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: api/hypershift/v1alpha1/nodepool_types.go [hunk 1]
```diff
@@ -823,6 +823,17 @@ type AWSNodePoolPlatform struct {
...
+	// Tenancy indicates if instance should run on shared or single-tenant hardware.
+	//
+	// Possible values:
+	// default: NodePool instances run on shared hardware.
+	// dedicated: Each NodePool instance runs on single-tenant hardware.
+	// host: NodePool instances run on user's pre-allocated dedicated hosts.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum:=default;dedicated;host
+	Tenancy string `json:"tenancy,omitempty"`
```
Comment (EmilienM) [inline api/hypershift/v1alpha1/nodepool_types.go:836]: You could create a new type (like AWSTenancy).
An example here:

https://github.com/openshift/hypershift/blob/d8812cdc1ea756b0857c13e0436ba8f4137a172f/api/hypershift/v1beta1/hostedcluster_types.go#L1952-L1974

And in the code you'll use the constants directly. Very handy and cleaner, imo.

Comment (muraee) [reply api/hypershift/v1alpha1/nodepool_types.go:836]: thanks for the suggestion @EmilienM.
This is field is just going to be passed through to CAPA as is, so we don't really need to use it's values directly.

Comment (EmilienM) [reply api/hypershift/v1alpha1/nodepool_types.go:836]: maybe something which can be done on CAPA side as well (just an idea).

**Pull request merged**